pvlib/iotools/tmy.py

```diff
@@ -150,24 +150,28 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
 
     References
     ----------
-
     .. [1] Wilcox, S and Marion, W. "Users Manual for TMY3 Data Sets".
        NREL/TP-581-43156, Revised May 2008.
-
     .. [2] Wilcox, S. (2007). National Solar Radiation Database 1991 2005
        Update: Users Manual. 472 pp.; NREL Report No. TP-581-41364.
-    '''
-
+    .. [3] `SolarAnywhere file formats
+       <https://www.solaranywhere.com/support/historical-data/file-formats/>`_
+    """  # noqa: E501
     head = ['USAF', 'Name', 'State', 'TZ', 'latitude', 'longitude', 'altitude']
-
-    with open(str(filename), 'r') as csvdata:
-        # read in file metadata, advance buffer to second line
-        firstline = csvdata.readline()
-        # use pandas to read the csv file buffer
-        # header is actually the second line, but tell pandas to look for
-        # header information on the 1st line (0 indexing) because we've already
-        # advanced past the true first line with the readline call above.
-        data = pd.read_csv(csvdata, header=0)
+    try:
+        fbuf = open(str(filename), 'r')
+    # SolarAnywhere files contain non-UTF8 characters and require
+    # encoding='iso-8859-1' on Linux in order to be parsed
+    except UnicodeDecodeError:
+        fbuf = open(str(filename), 'r', encoding='iso-8859-1')
```

Comment:
I believe it was the `readline()` call (not the `open()` call) that errors if the encoding is not correct.  Also if we're not using the context manager form of `open()` then best practice is to include an explicit `fbuf.close()`, preferably inside a `finally` clause, to ensure that open file handles aren't left lying around.

P.S. you currently have to re-tag the PR w/ `remote-data` to trigger those tests again.  That is something else to improve about the remote-data tests.  Sorry this PR is the guinea pig for the `remote-data` part of #1306!

Reply:
@kanderso-nrel Sadly you are correct. The try/except method gets a bit messier but I guess that's just how it is.

I think the remote-test failures are unrelated (`get_surfrad`). Also, I'm puzzled why four tests were canceled..